draft-ietf-rtcweb-mdns-ice-candidates.md

```diff
@@ -193,6 +196,33 @@ MUST expose a different mDNS name for each address.
 
 ### Implementation Guidance
 
+#### mDNS registration and query answer
+
+If the ICE agent skips both probing and announcement sending, the ICE agent
+does not send any mDNS message on the network. In that case, the ICE agent is
+expected to answer to any query matching the registered name.
+
+Sending an mDNS announcement will usually allow the ICE agent receiving the
+remote ICE candidate containing the mDNS name to already have the information
+useful to resolve the mDNS name if the mDNS ICE candidate can actually be used.
+If the mDNS announcement is not sent, this can increase the connection setup
+time.
+
+Sending the mDNS announcement to the network can be delayed, for instance due
```

Comment:
Are we trying to set down any normative requirements here?

Reply:
It seems that Chrome and WebKit implementation slightly varies in the way they might wait for registration of the candidate. The statement tries to say that you can wait or not, this is fine, since there is little harm in provide a candidate that cannot be resolved. I wonder whether the error case (thus not exposing a MDNS candidate) is somehow privacy problematic.

Reply:
Ideally we would take a SHOULD-strength position here (and for announcements), just to give clear guidance to implementors who don't want to have to think about all the details.

Reply:
> An ICE agent can provide the candidate to the web application before the mDNS announcement is sent on the network.

This is the key to minimize the impact of announcement on efficiency. I feel it would be ideal to make this point a SHOULD level, and an ICE agent SHOULD provide the candidate to the web application as soon as its mDNS name is generated, regardless of whether the announcement of this name is sent on the network.

Reply:
SHOULD as well